From: Helmut Grohne <helmut@subdivi.de>
Date: Mon, 20 Nov 2023 01:57:46 +0000
Subject: call pkgconfig in order to be able to cross build qtbase with MySql.
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=971604
Forwarded: not-needed
Reviewed-by: Lisandro Damián Nicanor Pérez Meyer <lisandro@debian.org>

Qt's build system calls mysql_config... which won't work in a cross build
environment like Debian's, as it will throw an exec format error.

In order to solve this call pkgconfig and use mysqlclient.pc.
---
 src/plugins/sqldrivers/configure.json | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/plugins/sqldrivers/configure.json b/src/plugins/sqldrivers/configure.json
index 9db93d6..5c35cd9 100644
--- a/src/plugins/sqldrivers/configure.json
+++ b/src/plugins/sqldrivers/configure.json
@@ -67,6 +67,7 @@
             },
             "headers": "mysql.h",
             "sources": [
+                { "type": "pkgConfig", "args": "mysqlclient" },
                 { "type": "mysqlConfig", "query": "--libs_r", "cleanlibs": true },
                 { "type": "mysqlConfig", "query": "--libs", "cleanlibs": true },
                 { "type": "mysqlConfig", "query": "--libs_r", "cleanlibs": false },
